bait-files/src/test/java/com/google/sps/UserTest.java

```diff
@@ -0,0 +1,131 @@
+package com.google.sps;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static com.google.sps.data.Keys.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import java.io.*;
+import javax.servlet.http.*;
+import javax.servlet.ServletException;
+import com.google.sps.servlets.UsernameTakenServlet;
+import com.google.sps.servlets.LoginServlet;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class UserTest {
```

Comment:
It's a good practice to create one unit test file per class you have and name them with the exact same name as the class you are testing except with the word "Test" added to the end.

So in your case, it looks like here you are testing the UsernameTakenServlet class and the LoginServlet, so I would say that you should create these files instead:
UsernameTakenServletTest.java  (containing class UsernameTakenServletTest)
LoginServlet.java  (containing class LoginServletTest)

Reply:
Separated concerns :)